### PR TITLE
Allow to convert to varchar an unsupported type in Phoenix

### PR DESCRIPTION
**Pull request by kokosing**
Allow to convert to varchar an unsupported type in Phoenix

### CODE REVIEW

**Comment by kokosing** [comment]
Fixes: #2426

**Comment by kokosing** [comment]
Fixes #2416

**Comment by ebyhr** [comment]
@kokosing Could you resolve conflicts?

**Comment by kokosing** [comment]
>@kokosing Could you resolve conflicts?

@ebyhr Done.

File: presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java [hunk 1]
```diff
@@ -272,7 +272,10 @@ public BaseJdbcClient(
                             .setComment(comment)
                             .build());
                 }
-                verify(columnMapping.isPresent() || getUnsupportedTypeHandling(session) == IGNORE, "Unsupported type handling is set to %s, but toPrestoType() returned empty");
```
Comment (ebyhr) [inline presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java]: PostgreSqlClient also has the same logic.

File: presto-phoenix/src/test/java/io/prestosql/plugin/phoenix/TestPhoenixIntegrationSmokeTest.java [hunk 2]
```diff
@@ -73,6 +76,36 @@ public void testMultipleSomeColumnsRangesPredicate()
...
+                "SELECT * FROM test_timestamp",
+                "VALUES " +
+                        "(1, null), " +
+                        "(2, '2002-05-30 09:30:10.500'), " +
+                        "(3, null)");
+        assertQueryFails(
+                withUnsupportedType(CONVERT_TO_VARCHAR),
+                "INSERT INTO test_timestamp VALUES (4, '2002-05-30 09:30:10.500')",
+                "Underlying type that is mapped to VARCHAR is not supported for INSERT: TIMESTAMP");
+        executeInPhoenix("DROP TABLE tpch.test_timestamp");
```
Comment (ebyhr) [inline presto-phoenix/src/test/java/io/prestosql/plugin/phoenix/TestPhoenixIntegrationSmokeTest.java]:
```suggestion
        assertUpdate("DROP TABLE test_timestamp");
```

**Comment by kokosing** [comment]
Comments addressed. @ebyhr Thanks for the review!

**Comment by ebyhr** [comment]
Unfortunately, it conflicted again.

**Comment by kokosing** [comment]
Rebased

**Pull request merged**